pkg/cli/client.go

```diff
@@ -70,8 +122,21 @@ func makeGraphQLRequest(url, query, username,
 }
 
 func doHTTPRequest(req *http.Request, verifyTLS bool, resultsPtr interface{}) (http.Header, error) {
-	if httpClient == nil {
-		httpClient = createHTTPClient(verifyTLS)
+	var httpClient *http.Client
+
+	host := req.Host
+
+	if httpClientsMap[host] == nil {
```

Comment:
I believe you should still use a mutex while reading/updating the dictionary.
There should not be a performance hit, most of the time waiting is for the request itself.